### PR TITLE
PAY-7200: Front end changes to Calculate the AMOUNT DUE and OVERPAYME…

### PR DESCRIPTION
**Pull request by Thor-tech-of-metal**
PAY-7200: Front end changes to Calculate the AMOUNT DUE and OVERPAYMENTS for ( Issue refund and Remission-Refunds)
           [PAY-7200] (https://tools.hmcts.net/jira/browse/PAY-7200).

### CODE REVIEW

File: projects/payment-lib/src/lib/components/case-transactions/case-transactions.component.ts [hunk 1]
```diff
@@ -561,6 +564,22 @@ export class CaseTransactionsComponent implements OnInit {
...
   }
 
+  calculateOverpaymentBaseOnAcceptedRefund() {
+    this.paymentGroups.forEach(paymentGroup => {
+      if (paymentGroup.refunds != null && paymentGroup.refunds.length > 0) {
+        console.log("aca1 !!!");
+
+        paymentGroup.refunds.forEach(refund => {
+          console.log("aca2 !!!"+ refund.refund_status.name);
+
```
Comment (Thor-tech-of-metal) [inline projects/payment-lib/src/lib/components/case-transactions/case-transactions.component.ts:574]: Nooooo I need to remove it. console.log("aca2 !!!"+ refund.refund_status.name);

Comment (Thor-tech-of-metal) [reply projects/payment-lib/src/lib/components/case-transactions/case-transactions.component.ts:572]: DONE al console  removed

**Review by davejones74 — APPROVED** [review]

**Review by Ollie-Smith — APPROVED** [review]
reviewed

**Review by davejones74 — DISMISSED** [review]

**Review by danlysiak — DISMISSED** [review]

**Review by Ollie-Smith — DISMISSED** [review]
A coding wizard here.  Amazing!

**Review by davejones74 — DISMISSED** [review]

**Review by chrisworthcgi — DISMISSED** [review]

**Review by davejones74 — CHANGES_REQUESTED** [review]

File: projects/payment-lib/src/lib/components/case-transactions/case-transactions.component.ts [hunk 2]
```diff
@@ -213,6 +215,7 @@ export class CaseTransactionsComponent implements OnInit {
           this.calculateAmountDueTo();
           this.calculateOverpayment();
```
[the remark this answers is not in the document]
Comment (Thor-tech-of-metal) [reply projects/payment-lib/src/lib/components/case-transactions/case-transactions.component.ts]: DONE

**Review by davejones74 — APPROVED** [review]

**Review by chrisworthcgi — APPROVED** [review]

**Review by Ollie-Smith — DISMISSED** [review]
reviewed

File: projects/payment-lib/src/lib/components/payment-view/payment-view.component.ts [hunk 3]
```diff
@@ -413,11 +413,22 @@ export class PaymentViewComponent implements OnInit {
...
-      return fee.add_remission && fee.remission_enable;
+      return fee.add_remission && fee.remission_enable && this.isTheCurrentRefundInProcessForThisFee(fee);
     } else {
       return false
     }
   }
+
+  // This method is going to check if the current refund has been rejected.
+  // If this is the case the button should be disable.
+  isTheCurrentRefundInProcessForThisFee(fee: IFee): boolean{
```
Comment (davejones74) [inline projects/payment-lib/src/lib/components/payment-view/payment-view.component.ts:424]: You have the same method defined twice in payment-vew.component.ts and service-request.component.ts? 

It's using the paymentLibComponent, so this needs to be added to the payment-lib.component.ts as you've duplicated the same code. 

You can then call :
paymentLibComponent.isTheCurrentRefundInProcessForThisFee(fee);

**Review by davejones74 — APPROVED** [review]

**Review by Ollie-Smith — COMMENTED** [review]
reviewed

**Review by Ollie-Smith — APPROVED** [review]
reviewed

**Review by davejones74 — APPROVED** [review]

**Review by chrisworthcgi — APPROVED** [review]

**Pull request merged**